src/main/scala/com/mozilla/telemetry/streaming/EventsToAmplitude.scala

```diff
@@ -0,0 +1,224 @@
+    val kafkaBroker: ScallopOption[String] = opt[String](
+      "kafkaBroker",
+      descr = "Kafka broker (streaming mode only)",
+      required = false)
+    val from: ScallopOption[String] = opt[String](
+      "from",
+      descr = "Start submission date (batch mode only). Format: YYYYMMDD",
+      required = false)
+    val to: ScallopOption[String] = opt[String](
+      "to",
+      descr = "End submission date (batch mode only). Default: yesterday. Format: YYYYMMDD",
+      required = false)
+    val fileLimit: ScallopOption[Int] = opt[Int](
+      "fileLimit",
+      descr = "Max number of files to retrieve (batch mode only). Default: All files",
+      required = false)
+    val raiseOnError:ScallopOption[Boolean] = opt[Boolean](
+      "raiseOnError",
+      descr = "Whether the program should exit on a data processing error or not.")
+    val failOnDataLoss:ScallopOption[Boolean] = opt[Boolean](
+      "failOnDataLoss",
+      descr = "Whether to fail the query when it’s possible that data is lost.")
+    val checkpointPath:ScallopOption[String] = opt[String](
+      "checkpointPath",
+      descr = "Checkpoint path (streaming mode only)",
+      required = false,
+      default = Some("/tmp/checkpoint"))
+    val startingOffsets:ScallopOption[String] = opt[String](
+      "startingOffsets",
+      descr = "Starting offsets (streaming mode only)",
+      required = false,
+      default = Some("latest"))
+    val url: ScallopOption[String] = opt[String](
+      "url",
+      descr = "Endpoint to send data to",
+      required = true)
+    val apiKey: ScallopOption[String] = opt[String](
+      "apiKey",
+      descr = "API Key to use for requests",
+      required = true)
+    val sample: ScallopOption[Double] = opt[Double](
+      "sample",
+      descr = "Fraction of clients to use",
+      required = false,
+      default = Some(1.0))
+    val minDelay: ScallopOption[Long] = opt[Long](
+      "minDelay",
+      descr = "Amount of delay between requesets in batch mode, in ms",
+      required = false,
+      default = Some(0))
+    val maxParallelRequests: ScallopOption[Int] = opt[Int](
+      "maxParallelRequests",
+      descr = "Max number of paralle requests in batch mode",
+      required = false,
+      default = Some(100))
+    requireOne(kafkaBroker, from)
+    conflicts(kafkaBroker, List(from, to, fileLimit, minDelay, maxParallelRequests))
+    validateOpt (sample) {
+      case Some(s) if 0.0 < s && s <= 1.0 => Right(Unit)
+      case Some(o) => Left(s"Sample out of range. Expected 0.0 < Sample <= 1.0, Found $o")
+    }
+
+    verify()
+  }
+
+  def parsePing(message: Message, sample: Double): Array[String] = {
+    implicit val formats = DefaultFormats
+
+    val fields = message.fieldsAsMap
+    val docType = fields.getOrElse("docType", "").asInstanceOf[String]
+    if (!allowedDocTypes.contains(docType)) {
+      throw new Exception("Doctype should be one of " + allowedDocTypes.mkString(sep = ","))
```

Comment:
I'm curious -- the default kafka topic is "telemetry", and this the firstdocType/appName check I see in the streaming path -- are they filtered before this?

Reply:
Nope, we don't know what the docType is until we unpack the protobuf, so it can't be done until this point. There has been talk of per-docType topics though.

Reply:
Then wouldn't the streaming version fail with an exception immediately?

Reply:
See line 140; we catch the exceptions.

Reply:
Hmm, I'm a bit torn on this -- it's not very idiomatic in JVM languages to catch expected exceptions, and it's supposed to be pretty slow to boot. I see the issue re: handling batch vs streaming exceptions differently, but I'd advocate for adding a check on whether we're running in a streaming context or not before throwing an exception as opposed to catching upstream.

In the process of doing that, you'd probably remove the code to catch Throwables, but for future reference: https://www.sumologic.com/blog/code/why-you-should-never-catch-throwable-in-scala/

Reply:
Good point. We should change this in ErrorAggregator as well. The downside is there are a whole host of errors that can arise during parsing, and it's nigh impossible to decipher what they are and enumerate those during testing. Let me consider what to do here.

Reply:
@sunahsuh I've avoided catching expected exceptions, but kept the catch-all try. I'm thinking we circle back to this later and fix it in both this code and ErrorAggregator.